orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTask.groovy

```diff
@@ -178,12 +180,36 @@ class MonitorKatoTask implements RetryableTask, CloudProviderAware {
         log.error("Request failed attempting to resume task", e)
       }
       status = ExecutionStatus.RUNNING
+
+      outputs["kato.task.terminalRetryCount"] =
+        ((stage.context."kato.task.terminalRetryCount" as Integer) ?: 0) + 1
+
       stage.context."kato.task.retriedOperation" = true
+
+      if (katoTask.resultObjects?.find { it.type == "EXCEPTION" }) {
+        def exception = katoTask.resultObjects.find { it.type == "EXCEPTION" }
+        log.debug("Retrying kato task ${katoTask.id} with exception: ", exception)
+      } else {
+        log.debug("Retrying kato task ${katoTask.id}")
+      }
     }
 
     TaskResult.builder(status).context(outputs).build()
   }
 
+  private boolean terminalRetryCondition(Stage stage, Task katoTask, ExecutionStatus status) {
+    return (
+      status == ExecutionStatus.TERMINAL
+      && katoTask.status.retryable
+      && dynamicConfigService.isEnabled("tasks.monitor-kato-task.terminal-retries", true)
+      && (stage.context."kato.task.terminalRetryCount" as Integer) < dynamicConfigService.getConfig(
```

Comment:
Instead of `tasks.monitor-kato-task.saga-retries` I change this to `tasks.monitor-kato-task.terminal-retries` as I didn't want Orca referring to anything implementation specific on the CloudDriver side.